doofinder.php

```diff
@@ -70,7 +70,13 @@ public function install()
             && $this->registerHook('displayHeader')
             && $this->registerHook('moduleRoutes')
             && $this->registerHook('actionProductSave')
-            && $this->registerHook('actionProductDelete');
+            && $this->registerHook('actionProductDelete')
```

Comment:
@ivan-doofinder una pregunta, por qué para productos tenemos `productSave` y `productDelete` y para `cms` y `category` está `add`, `update` y `delete`? y el sufijo `After` que significa que sea después de que se guarde?

Reply:
Para el caso de cms y category se usan los hooks de la clase ObjectModel: [ObjectModel.php#L628](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/classes/ObjectModel.php#L628).
`Hook::exec('actionObject' . $this->getFullyQualifiedName() . 'AddAfter', ['object' => $this]);`

Existe también el hook que se ejecuta antes de la modificación, de ahí la opción de after/before: [ObjectModel.php#L557](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/classes/ObjectModel.php#L557)
`Hook::exec('actionObject' . $this->getFullyQualifiedName() . 'AddBefore', ['object' => $this]);`

El producto sin embargo, tiene hooks propios definidos [Product.php#L830](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/classes/Product.php#L830). Y para este caso, el **actionProductSave** se ejecuta tanto en el add como en el update.
`Hook::exec('actionProductSave', ['id_product' => (int) $this->id, 'product' => $this]);`